ENS/ens-beginners-guide.md

```diff
@@ -0,0 +1,222 @@
+# ENS: Beginner's Guide
+
+User experience is a key challenge when developing a dapp. Ethereum has complexities that should be hidden away to enable broad adoption. One such complexity, is an Ethereum address, it is long, unwieldy, and hard to remember.
+
+The Ethereum Name Service (ENS) is an incredibly useful tool for dapp developers. ENS is like DNS, in that it maps a memorable shortcut to an address. Using ENS we can map the friendly name `ethereum.eth` to the rather unfriendly `0xfB6916095ca1df60bB79Ce92cE3Ea74c37c5d359`. Subsequently, the friendly name can be used in-place of the address, making it easier to remember, and reducing the chance of errors. 
+
+An ENS name has a root name `ethereum.eth` which can contain sub-names like `wallet.ethereum.eth`. The root name is owned by whoever successfully bid for it in a [Vickrey auction](https://medium.com/the-ethereum-name-service/a-beginners-guide-to-buying-an-ens-domain-3ccac2bdc770 "Vickrey auction").
+
+In fact, ENS doesn't just map to addresses. ENS is highly extensible and supports many types of mappings. The same friendly name can be mapped to multiple endpoints at the same time. 
+
+So the `ethereum.eth` name could point to:
+- A multisig wallet contract address
+- Public encryption keys for secure communication
+- Website content (via IPFS hash or multihash)
+
+## Code Walkthrough
+
+Many Ethereum libraries, including Web3.js now support ENS lookups out-of-the-box. Unfortunately using ENS is not as intuitive as the library documentation suggests; there are a number of things to look out for.
+
+To demonstrate how to lookup ENS names we will write a simple Node.js command-line tool called `enslookup`. Given an ENS name as an argument it will query the ENS registry for information about that name.
+
+### Prerequisites
+
+First you will need to install Node.js and npm by following the install instructions on the [Node.js website](https://nodejs.org/ "Node.js").
+
+If you are not running a full mainnet Ethereum node, you will need to setup an Infura endpoint for connecting. This is easy to do - just sign up on the [infura.io](https://infura.io) website. Create a new project and copy the mainnet URL. It will be something like this: `mainnet.infura.io/v3/<your project id>`.
```

Comment:
Future tense is another odd one to understand, this isn't really in the future, you're doing it now, so try:

```suggestion
If you are not running a full mainnet Ethereum node, you need to setup an Infura endpoint for connecting. This is easy to do - just sign up on the [infura.io](https://infura.io) website. Create a new project and copy the mainnet URL. It will be something like this: `mainnet.infura.io/v3/<your project id>`.
```

Reply:
Updated and reviewed text for more future tense